src/constants/triggers.ts

```diff
@@ -0,0 +1,89 @@
+export enum ResourceType {
+  TASK = "TASK",
+  VERSION = "VERSION",
+  BUILD = "BUILD",
+}
+
+type PayloadResourceIdKey = "in-version" | "in-build" | "id";
+export interface Trigger {
+  trigger: string;
+  label: string;
+  extraFields?: ExtraField[];
+  resourceType: ResourceType;
+  payloadResourceIdKey?: PayloadResourceIdKey;
+  regexSelectors?: RegexSelector[];
+}
```

Comment:
These should probably be defined in a types file since these are typescript types

Reply:
done